File: 101.SymmetricTree/clywin123/isSymmetric.c
bool isSameTree(struct TreeNode* p, struct TreeNode* q) {
	if(p==NULL || q ==NULL){
		if(p==NULL && q==NULL)	
			return true;
		else{
			return false;
		}
	}
	else if(p->val!=q->val){
		return false;
	}
	else
		return isSameTree(p->left,q->left)& isSameTree(p->right,q->right);
}

struct TreeNode* invertTree(struct TreeNode* root) {
	struct TreeNode *tmp;
	if(root){
		tmp = root -> left;
		root -> left = root -> right;
		root -> right = tmp;
		invertTree(root->left);
		invertTree(root->right);
	}
	return root;
}

struct TreeNode* copy(struct TreeNode* root){
	struct TreeNode *node;
	if(root == NULL)
		return NULL;
	if(root){
		node = (struct TreeNode*)malloc(sizeof(struct TreeNode));
		node->val = root->val;
		node->left= copy(root->left);
		node->right= copy(root->right);
	}
	return node;
}

bool isSymmetric(struct TreeNode* root) {
	struct TreeNode *tmp;
	tmp = copy(root);
	invertTree(root);
	return isSameTree(root,tmp);
}

File: 101.SymmetricTree/wyrj/isSymmetric.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    const f = (lhs, rhs) => {
        if (!lhs || !rhs) return lhs === rhs;
        return lhs.val === rhs.val && f(lhs.left, rhs.right) && f(lhs.right, rhs.left);
    }
    return f(root, root);
};
